frontend/apps/marketing/src/components/carousels/actionBlockCarousel/ActionBlockCarousel.tsx

```diff
@@ -0,0 +1,133 @@
+'use client';
+
+import {BaseEntry, EntryFields} from 'contentful';
+import '@code-dot-org/component-library/carousel/index.css';
+import React, {ReactNode, useMemo} from 'react';
+
+import ActionBlock, {
+  ActionBlockProps,
+} from '@code-dot-org/component-library/actionBlock';
+import DSCOCarousel from '@code-dot-org/component-library/carousel';
+
+export type ActionBlockCarouselProps = ActionBlockProps & {
```

Comment:
Aside from other comments, I think it's a good place to use our new generic type `ContentfulEntry` you can find it here,  just pull the latest staging https://github.com/code-dot-org/code-dot-org/blob/staging/frontend/apps/marketing/src/contentful/types/entries/ContentfulEntry.ts

here's example how it can be used - https://github.com/code-dot-org/code-dot-org/blob/staging/frontend/apps/marketing/src/contentful/types/entries/ImageAssetEntry.ts

(keep using EntryFields types for field values types, that is awesome!)

Reply:
Updated, thanks for pairing!